fahdi/css-visual-learner
# Implement :nth-child visualizer and update related styles

This Pull Request introduces an interactive `:nth-child` visualizer to the CSS Visual Learner tool. It allows users to understand how the `:nth-child` pseudo-class works in a practical, visual manner. The visualizer supports dynamic input, providing immediate visual feedback as the user types in different `:nth-child` patterns.

#### What's included in this PR:

1. **New Visualizer Page**: Added `nth-child.html`, a dedicated page for the `:nth-child` visualizer feature, enhancing our educational toolkit.

2. **User Input Interaction**: Developed `scripts/nth-child.js` to handle user input, allowing for real-time updates to list item styles based on the provided `:nth-child` pattern.

3. **Styling Enhancements**: Updated `style.css` to include default styles for list items, showcasing the use of `:nth-child` in a visually engaging default state.

4. **Flexbox Instruction Update**: Modified the Flexbox example within `script.js` to prompt users to explore all Flexbox properties via a dedicated menu, encouraging further learning and interaction.

#### Purpose of This Feature:

The `:nth-child` visualizer serves to:
- Provide an interactive learning experience for users new to CSS pseudo-classes.
- Offer a tool for experienced developers to quickly test and visualize `:nth-child` patterns.
- Enhance our suite of CSS educational resources, contributing to a more comprehensive learning platform.

#### Testing:

Thorough testing has been conducted to ensure cross-browser compatibility and responsiveness. The dynamic input field has been tested with a variety of `:nth-child` patterns to verify the visualizer's functionality.

#### Screenshots/Demos:

[Optional: Include any relevant screenshots or video demos that illustrate the changes made.]

#### How to Test This Feature:

1. Navigate to the `nth-child.html` page.
2. Enter various `:nth-child` patterns into the input field. like `1`, `n+1`, `2n+1` etc
<img width="1063" alt="Screenshot 2023-11-23 at 8 16 54 AM" src="https://github.com/fahdi/css-visual-learner/assets/434134/4e547bbd-80b6-4564-9cef-9f5ca7a83db7">

<img width="1109" alt="Screenshot 2023-11-23 at 8 16 45 AM" src="https://github.com/fahdi/css-visual-learner/assets/434134/22790cbb-5c0a-4750-a582-da4309f3bc87">
4. Observe the changes in the list item styles in real-time.
